src/library/dao/trackdao.cpp

```diff
@@ -1486,12 +1499,16 @@ TrackPointer TrackDAO::getTrackById(TrackId trackId) const {
 
     // For every column run its populator to fill the track in with the data.
     bool shouldDirty = false;
-    for (int i = 0; i < recordCount; ++i) {
-        TrackPopulatorFn populator = columns[i].populator;
-        if (populator != nullptr) {
-            // If any populator says the track should be dirty then we dirty it.
-            if ((*populator)(queryRecord, i, pTrack.get())) {
-                shouldDirty = true;
+    {
+        int recordCount = queryRecord.count();
+        VERIFY_OR_DEBUG_ASSERT(recordCount == columnsCount) {
+            recordCount = math_min(recordCount, columnsCount);
+        }
+        for (int i = 0; i < recordCount; ++i) {
+            TrackPopulatorFn populator = columns[i].populator;
+            if (populator) {
+                // If any populator says the track should be dirty then we dirty it.
+                shouldDirty |= (*populator)(queryRecord, i, pTrack.get());
```

Comment:
Nitpick: `shouldDirty` is `bool`, so using the bitwise operator feels like a hack. Unfortunately there is no `||=` operator, and `shouldDirty = shouldDirty || ...` is a bit long :(

Reply:
I am aware that the bitwise operator is not ideal. Just wanted to get rid of the `if` statements that is even more verbose and unnecessarily complex.

Then let's use `= ||` as you proposed. Any expression is better than a statement.

Reply:
Ouch, that's tricky. shouldDirty = shouldDirty || does not work because of short circuit evaluation.

Reply:
https://github.com/mixxxdj/mixxx/pull/4633/commits/e6eb30c29bda8b6ddee7639a6292ba67f826063e